packages/web-app-files/src/components/AppBar/AppBar.vue

```diff
@@ -163,17 +162,14 @@ export default {
     ContextActions
   },
   mixins: [Mixins, MixinFileActions],
-  setup() {
-    const router = useRouter()
-    return {
-      isPersonalLocation: isLocationSpacesActive(router, 'files-spaces-personal-home'),
-      isPublicLocation: isLocationPublicActive(router, 'files-public-files')
-    }
-  },
   data: () => ({
     newFileAction: null,
     path: '',
-    fileFolderCreationLoading: false
+    fileFolderCreationLoading: false,
+    isPersonalLocation: false,
```

Comment:
What was the reason for switching to `data` props instead of using the refs from the setup function?

Reply:
Previously, it was only being set in `setup()`. This lead to this property being read on page load, but not on router changes. Thus we played around a bit, moved those properties to `data` and eventually ended up having a watcher on route changes. So I believe we can move those back to `setup()`.

Is having such things in `setup` the preferred way? Or how do we/Vue handle this?

Reply:
Uff. Yes, preferred to have it in `setup()`, but really seems to have broken reactivity. Could you build them as computed props instead of data+watcher? Reactivity in setup() can be fixed in a separate PR. Is also the issue here: https://github.com/owncloud/web/issues/6326

Reply:
we tried to do computed, but that didn't seem to work...

Reply:
I implemented another helper that watches the current route internally and can be used like this: https://github.com/owncloud/web/blob/65f0a30fe01f1b8aae2f4721c58134c3297d5e5b/packages/web-app-files/src/components/AppBar/AppBar.vue#L172 

Could you switch back to the properties in the `setup` function like shown in the AppBar?